tomcat/src/main/java/org/apache/coyote/http11/HttpRequestLine.java

```diff
@@ -0,0 +1,51 @@
+package org.apache.coyote.http11;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.coyote.exception.UncheckedHttpException;
+
+public class HttpRequestLine {
+
+    private static final String REQUEST_LINE_DELIMITER = " ";
+    private static final int REQUEST_LINE_COUNT = 3;
+
+    private final HttpMethod httpMethod;
+    private final HttpRequestUri requestUri;
+    private final String httpVersion;
+
+    public HttpRequestLine(String input) {
+        validateNotNull(input);
+        String[] splitLine = input.split(REQUEST_LINE_DELIMITER);
+        validateLineSize(splitLine);
+        this.httpMethod = HttpMethod.valueOf(splitLine[0]);
+        this.requestUri = HttpRequestUriParser.parse(splitLine[1]);
+        this.httpVersion = splitLine[2];
+    }
+
+    private void validateNotNull(String input) {
+        if (input == null) {
+            throw new UncheckedHttpException(new IllegalArgumentException("Http Request Line은 비어있을 수 없습니다."));
+        }
+    }
+
+    private void validateLineSize(String[] splitLine) {
+        if (splitLine.length != REQUEST_LINE_COUNT) {
+            throw new UncheckedHttpException(new IllegalArgumentException("Http Request Line 형식이 잘못 되었습니다."));
+        }
+    }
+
+    public HttpResponse<String> getHttpResponse(Map<String, String> bodies) throws IOException {
+        if (httpMethod.isGet()) {
+            HttpResponse<?> httpResponse = requestUri.processParams(httpMethod);
+            return requestUri.getHttpResponse(httpResponse);
+        } else if (HttpMethod.POST.equals(httpMethod)) {
+            HttpResponse<?> httpResponse = requestUri.processParams(httpMethod, bodies);
+            return requestUri.getHttpResponse(httpResponse);
+        }
+        throw new UncheckedHttpException(new UnsupportedOperationException("지원하지 않는 기능입니다."));
+    }
```

Comment:
이미 if문 내부에서 return을 하기 때문에 else if문을 사용하지 않아도 될 것 같아요!
```suggestion
        if (httpMethod.isGet()) {
            HttpResponse<?> httpResponse = requestUri.processParams(httpMethod);
            return requestUri.getHttpResponse(httpResponse);
        } 
        if (HttpMethod.POST.equals(httpMethod)) {
            HttpResponse<?> httpResponse = requestUri.processParams(httpMethod, bodies);
            return requestUri.getHttpResponse(httpResponse);
        }
        throw new UncheckedHttpException(new UnsupportedOperationException("지원하지 않는 기능입니다."));
    }
```

Reply:
Early Return이기에 분기문 간 서로 간섭은 안하겠네요.
감사해요 배키!